LayoutTests/fast/forms/color/display-none-input-color-chooser-shown.html

```diff
@@ -0,0 +1,35 @@
+<!DOCTYPE html>
+<html>
+<head>
+<script src="../../../resources/ui-helper.js"></script>
+<script src="../../../resources/testharness.js"></script>
+<script src="../../../resources/testharnessreport.js"></script>
+</head>
+<body>
+<input id="colorPick" type="color" />
+<label for="colorPick" id="labelPick">Pick a color</label>
+<script>
+    if (window.testRunner) {
+        testRunner.dumpAsText();
+        testRunner.waitUntilDone();
+    }
+    async function runTest() {
+        if (!testRunner.runUIScript)
+            return;
+            
+    test(function() {
+    var colorPicker = document.getElementById("colorPick");
+    colorPicker.style.display = "none";
+    var labelPick = document.getElementById("labelPick");
+    var x = labelPick.offsetLeft + labelPick.offsetWidth/2;
+    var y = labelPick.offsetTop + labelPick.offsetHeight/2;
+    eventSender.mouseMoveTo(x, y);
+    eventSender.mouseDown();
+    eventSender.mouseUp();
+    
+    assert_true(UIHelper.activateElement(input));
+    assert_true(UIHelper.ensurePresentationUpdate());
+    testRunner.notifyDone();
```

Comment:
Does this test fail without your change? This doesn't actually assert that a color picker is presented.

I think we need a new testing hook like `UIHelper.isShowingColorPicker()`, similar to `isShowingDataListSuggestions()`.

Reply:
Yes - it fails without my change saying that it does not recognise "testRunner.isChooserShown", which reflects that we  have same architecture as of Chrome. Hence, I started using UIHelper but not quite good in writing test cases yet. So struggling bit. Do we have any Webkit reference guideline? Thanks!

Reply:
Right, WebKit does not have an equivalent for `testRunner.isChooserShown`. We need to add something similar to `isShowingDateTimePicker`, as done in this [patch](https://github.com/WebKit/WebKit/commit/f306f45ea9acd28b9c8b767f3d01e545297e7a7c).

The testing hook is going to be platform-specific, so we would need to skip the test on platforms where we are unable to implement.

I could add the hooks in a separate patch if that would be helpful, or let me know if you would like to try implementing them yourself.